tritium-metrics/src/main/java/com/palantir/tritium/metrics/MetricName.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics;
+
+import java.util.Map;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Value.Style(
+        jdkOnly = true,
+        get = {"get*", "is*"},
+        overshadowImplementation = true,
+        visibility = Value.Style.ImplementationVisibility.PACKAGE)
+public interface MetricName {
+    String name();
+    Map<String, String> tags();
```

Comment:
I know we talked about this, but probably worth JavaDoc is the assumption that both tag name and value are assumed to be `@Safe` loggable

Reply:
done